designer/client/src/components/modals/ValidationLabels.tsx

```diff
@@ -29,22 +27,42 @@ export default function ValidationLabels(props: Props) {
 
     const renderErrorLabels = () =>
         validationErrors.map((validationError, ix) => (
-            <span key={ix} className="validation-label-error" title={validationError.description}>
-                {validationError.message}
-            </span>
+            // we don't pass description as tooltip message - we pass the entire non-line-limited string until we make
+            // changes on the backend
+            <LimitedValidationLabel key={ix} message={validationError.message} tooltipMessage={validationError.message} type={"ERROR"} />
         ));
 
     // TODO: We're assuming that we have disjoint union of type info & validation errors, which is not always the case.
     // It's possible that expression is valid and it's type is known, but a different type is expected.
     return (
-        <div className={`validation-labels ${additionalClassName}`}>
+        <div className={`validation-labels`}>
             {isValid ? (
-                <span className="validation-label-info" title="Info">
-                    {validationLabelInfo}
-                </span>
+                <LimitedValidationLabel message={validationLabelInfo} tooltipMessage={validationLabelInfo} type={"INFO"} />
             ) : (
                 renderErrorLabels()
             )}
         </div>
     );
 }
+
+type ValidationLabelProps = {
+    message: string;
+    tooltipMessage: string;
+    type: ValidationLabelType;
+};
+
+type ValidationLabelType = "INFO" | "ERROR";
+
+const labelTypeMap = {
+    INFO: "validation-label-info",
+    ERROR: "validation-label-error",
+};
+
+function LimitedValidationLabel(props: ValidationLabelProps): JSX.Element {
+    const className = `${labelTypeMap[props.type]} line-cut`;
```

Comment:
prefer removing hardcoded, not connected, global classnames

Reply:
done